### DISCUSSION TITLE
Run jest testing for components using wavesufer.js

### DISCUSSION THREAD

**Discussion by hieunm22** [katspaugh/wavesurfer.js#3482] [Q&A] [answered]
I'm using jest to run test my component
Everything looks good until I added this library to my component to replace built-in html audio tag
The error thrown in the console looks like this

``` typescript
FAIL  src/pages/Home/Messages/MessageItem.test.tsx
  ● Test suite failed to run

    ReferenceError: self is not defined

      2 | import usePrevious from "../../hooks/usePrevious"
      3 | import classnames from "classnames"
    > 4 | import WaveSurfer from "wavesurfer.js"
        | ^
      5 | import { AudioWithWaveProps } from "./AudioWithWave.types"
      6 | 

      at Object.<anonymous> (node_modules/wavesurfer.js/dist/webpack:/WaveSurfer/webpack/universalModuleDefinition:10:2)
      at Object.<anonymous> (src/components/AudioWithWave/AudioWithWave.hook.ts:4:1)
```

I'm wondering is it a bug or I missed something extra when adding wavesufer.js to my component
Thanks for your attentions

**Comment by katspaugh** [comment]
Looks like your build is trying to import a minified browser file instead of an ESM module. Try specifying a full path: `wavesurfer.js/dist/wavesurfer.esm.js`

Also could you post your build config? In the tsconfig it’s usually the moduleResolution param that has to be adjusted.

**Reply by hieunm22** [reply]
hi, here's my tsconfig

```json
{
  "compilerOptions": {
    "target": "es5",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "allowJs": false,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noFallthroughCasesInSwitch": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react-jsx"
  },
  "include": [
    "src"
  ]
}
```

**Reply by katspaugh** [reply]
Yeah, tnx, I think `"moduleResolution": "node",` would try to load a CommonJS module instead of an ES module. Try changing it to `esnext` and it should work.

**Reply by hieunm22** [reply, marked as answer]
I don't know why but your suggestion doesn't work. I installed both `wavesufer.js` and `@types/wavesurfer.js`, but put they in the same section in package.json: `dependencies`. When I move `@types/wavesurfer.js` to `devDependencies`, upgrade `wavesufer.js` from `6.2.0` to `^7.2.3`, it works fine. Just a little bit update in `package.json` only
Anyway, thank you for your reply

**Reply by katspaugh** [reply]
Yes, exactly, `@types/wavesurfer.js` should not be installed for v7, it's an old package not maintained by me that is not needed anymore. Please see the [readme](https://github.com/katspaugh/wavesurfer.js#getting-started).
